mlflow/pyfunc/__init__.py

```diff
@@ -2957,7 +2958,11 @@ def predict(self, context, model_input: List[str], params=None) -> List[str]:
             _logger.info("Predicting on input example to validate output")
             context = PythonModelContext(artifacts, model_config)
             python_model.load_context(context)
-            output = python_model.predict(context, messages, params)
+            if inspect.signature(python_model.predict).parameters.get("context"):
+                _logger.info(_DROP_CONTEXT_IN_CHAT_MODEL_PREDICT_INFO)
```

Comment:
Can we use `warnings.warn`?

Reply:
Should this be a warning?

Reply:
ah I thought this is a deprecation warning.

Reply:
I think we can add a deprecation warning before we plan to drop it completely in our implementation, but for now it's more like 'letting users aware this is also supported'